library.rb

```diff
@@ -1,3 +1,119 @@
+require './author'
+require './book'
+require './reader'
+require './order'
+
 class Library
+  REQUIRED_COLLECTIONS = %w(Author Book Reader Order)
+  attr_reader :authors, :books, :readers, :orders
+  def initialize
+    @authors = []
+    @books = []
+    @readers = []
+    @orders = []
+    puts 'Use :load method for load library from file.'
+  end
+
+  def load
+    REQUIRED_COLLECTIONS.each do |name|
+      load_from_file(name)
+    end
+  end
+
+  def save
+    REQUIRED_COLLECTIONS.each do |name|
+      save_to_file(name)
+    end
+  end
+
+# for interface
+  def new_author(params)
+    new_class_instance("Author", @authors, params)
+  end
+
+  def new_book(params)
+    if !@authors.map(&:name).index(params.last)
+      puts "Library does not have the author #{params.last}"
+      return
+    end
+    new_class_instance("Book", @books, params)
+  end
+
+  def new_reader(params)
+    new_class_instance("Reader", @readers, params)
+  end
+
+  def new_order(params)
+    if !@books.map(&:title).index(params.first)
+      puts "Library does not have the book #{params.first}"
+      return
+    end
+    if !@readers.map(&:name).index(params[1])
+      puts "#{params[1]} wasn't registered at this library"
+      return
+    end
+    new_class_instance("Order", params)
+  end  
+
+  def best_reader
+    readers = @orders.map(&:reader)
+    uniq_readers = readers.uniq
+    uniq_readers.sort! { |x,y| readers.count(y) <=> readers.count(x) }
+    # in case when we have few readers with equel quantity read books
+    uniq_readers.select { |e| readers.count(e) == readers.count(uniq_readers.first) }
```

Comment:
Please give an example of order. I think that you do a lot of extra things. I will continue review after.